janitor/functions/pivot.py

```diff
@@ -648,13 +713,14 @@ def _pivot_longer_dot_value(
                 "Kindly use a unique name."
             )
     if index:
-        exclude = set(index).intersection(exclude)
-        if exclude:
-            raise ValueError(
-                f"Labels {*exclude, } already exist "
-                "as column labels assigned to the dataframe's "
-                "index parameter. Kindly use a unique name."
-            )
+        for word in index:
+            if word in exclude:
+                raise ValueError(
+                    f"{word}  associated with .value "
+                    "already exists as a column label "
+                    "assigned to the dataframe's index parameter. "
+                    "Kindly use a unique name."
+                )
```

Comment:
My thought process was trigger error as early as possible. What's d pattern for these? All errors at once?

Reply:
Hmmm, good points, @thatlittleboy and @samukweku. 

I think I can see why triggering as early as possible would be good, especially in the case where it takes a bit of time (~seconds to ~minutes) to collate together all of the errors. Raising early gives a bit more interactive feel.

Thinking about the user experience with `pandera`, there's perhaps something to learn there -- any rows that do not satisfy pandera checks are collected together and displayed at once.

If I were to vote, it would be:

- raise all errors all at once if the code chunk runs really fast
- raise the first error while also letting users know, in the error message, that "we raise on first error, may be more" is the behaviour of the error handling.

Reply:
@samukweku could I check, is this issue resolved, btw? I just realized I commented on an outdated comment thread :joy_cat:.

Reply:
Sorry for the late reply. Yeah, there is definitely a place for both patterns (raising early / collating errors as a group and raising all at once).

I would tend to agree with Eric's general intuition (using "code runtime" as a rough gauge); For example, import errors are typically collated together (e.g. this function requires optional import `xxx`, `yyy` and `zzz`, please pip install them. [Pandas](https://github.com/pandas-dev/pandas/blob/main/pandas/__init__.py#L17) does this.). You can imagine how annoying it might be if this information gets relayed to you one package at a time.

In our particular case, we're essentially checking if there is an intersection between the two sets of names `index` and `exclude` (which was how the code was written before this PR). Not particularly computationally expensive I think, if that's the primary worry here.
The only instance you'll gain some time advantage is when there *is* an intersection (ala, raising the error early), but the trade-off in terms of user experience is too huge, IMO. 

Not to say there aren't situations where raising early is useful -- there are --; just not in this particular case, I think.

Reply:
thanks for the feedback; I made the changes to collate the errors into one; I'll see if there are any other error sections i can lump into a one time call

Reply:
Thanks, approving and merging!